### DISCUSSION TITLE
Best way to update CashFactory

### DISCUSSION THREAD

**Discussion by Bidmaster777** [OlivierGaland/CashFactory#26] [Q&A] [answered]
Hello, I was wondering what is the best way to update an existing install of CashFactory?? Simply follow the steps again?

**Comment by OlivierGaland** [comment, marked as answer]
Hello,

If you want to install a new release, the "brainless" procedure is :  
- stopping the stack : sudo docker-compose down
- back up your .env file (and possibly docker-compose.yml if you have special tuning inside)
- delete the cashfactory dir and apply installation procedure : 
              - download new release and unzip
              - No need to run setup.sh again (docker already installed) or create account on apps (except if a new app was added)
              - backport your .env settings in .env file (usually a simple copy will be enough) and possibly your docker-compose tuning
              - start the stack : sudo docker-compose -d up

! be warned that the earnapp container may have changed its id, for this case you need to start earnapp and get the new id (earnapp showid) and go to the following url to register it to your account : https://earnapp.com/r/your-id replacing "your-id" with the id you found earlier. Other applis should work without any changes.

The best way is keep your stack running and update accordingly to new changes, changes often impact only one container so no need to restart/destroy all. This need a little more check and no standard procedure can be written.

For instance if the webserver has been changed, you need to copy the new webserver file in your directory, on portainer delete the webserver container then build/start it with this command :  sudo docker-compose up -d --no-deps --build Webserver

Release 0.5 added a new tool : Traffmonetizer, in that case, you need to update .env file to add the credential, get the new docker-compose.yaml file that contain the settings for the added service and run the same command to build/start the new container : sudo docker-compose up -d --no-deps --build TraffMonetizer

Those are just docker and docker-compose command, you may have a look at a few tutorials to see how it works. Portainer is here only because it is a friendly GUI to manage then, but the CLI command will also works.

Olivier
